### DISCUSSION TITLE
Algorithm used to create the microstructure in Neper

### DISCUSSION THREAD

**Discussion by kumaranvenk** [neperfepx/neper#818] [Q&A] [answered]
Hi, I have small doubt regarding generating microstructure in neper.

When I am using the calculated weights to generate the microstructure it is giving me the ordinary voronoi, which was not similar to the actual image. However, when I find the smallest area among all grains and then divided the areas of all the grains with the smallest area. Then i used the data as weight, it gave me the desired result.

Can anyone elaborate on the algorithm used, is it (euclidean distance)^2-weigth as given the paper **_"The Neper/FEPX Project: Free / Open-source Polycrystal Generation, Deformation Simulation, and Post-processing"_** or something else

**Comment by rquey** [comment, marked as answer]
See https://neper.info/papers.html#id2 for more details on the method. I think you are looking for [-morpho centroidsize](https://neper.info/doc/neper_t.html#cmdoption-morpho).
